docs/pages/guides/send-event-in-server.md

```diff
@@ -0,0 +1,111 @@
+- `WithCategory` : The string argument to specify under which category this event falls, For example, when a user is provisioning a Kubernetes connection, the specified category for the event is "connection.
+- `WithAction` : The string argument to specify the type of action the server is performing, for instance, when registering a Kubernetes connection, the action would be "register".
+
+*Note: categories and actions must be in snake case,*
+      example categories: **"signup_request", "github_artifacts"**.
+      example actions: **"remove_from_organization", "add_to_organization"**.
+
+- `WithSeverity`: it takes [EventSeverity](https://github.com/meshery/meshkit/blob/ea3c60907a1cd1902902a4113206579992772083/models/events/events.go#L75) (underlying type is string), to specify severity of the event to bring user's attention to the event accordingly.
+- `WithDescription` : The string argument provides a detailed, descriptive message that depicts the specific operation or action being executed on the server.
+- `WithMetadata`: it takes a Map `map[string]interface{}` data structure containing any supplementary information that the developer/contributor deems essential for the user to be informed about.
+- `Build` : returns the [Event](https://github.com/meshery/meshkit/blob/ea3c60907a1cd1902902a4113206579992772083/models/events/events.go#L37) instance constructed upon the previously described functions and prepares it for publication through the [Broadcast](https://github.com/meshery/meshery/blob/1b5d78ed34648e0a91df8c2273026b930f748fbc/server/models/event_broadcast.go#L14), which is responsible for disseminating events.
+
+
+
+## An Example in Code
+
+<pre class="codeblock-pre">
+ <div class="codeblock">
+func (h *Handler) KubernetesPingHandler(w http.ResponseWriter, req *http.Request, _ *models.Preference, user *models.User, provider models.Provider) {
+	userID := uuid.FromStringOrNil(user.ID)
+	token, ok := req.Context().Value(models.TokenCtxKey).(string)
+	if !ok {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "failed to get the token for the user")
+		return
+	}
+
+	connectionID := req.URL.Query().Get("connection_id")
+	eventBuilder := events.NewEvent().FromUser(userID).FromSystem(*h.SystemID).WithCategory("connection").WithAction("ping")
+	if connectionID != "" {
+		// Get the context associated with this ID
+		k8sContext, err := provider.GetK8sContext(token, connectionID)
+		eventBuilder.ActedUpon(uuid.FromStringOrNil(connectionID))
+		if err != nil {
+			eventBuilder.WithSeverity(events.Error).WithDescription(fmt.Sprintf("Encountered an issue while retrieving kubernetes context for connection ID `%s`", connectionID)).
+				WithMetadata(map[string]interface{}{
+					"error":                 err,
+					"probable_cause":        "Meshery might not be able to access Kubeconfig.",
```

Comment:
This is incorrect @zakisk, the severity of the event is an error, therefore the error should be MeshKit error, and the error would contain `probable_cause` and `suggested_remediation`.

Please update the example. Or add a snippet/link from the existing codebase.
When you add a link make sure to not source from the master branch but instead from one of the release tags. As in future, there will be changes in the code, which would result in invalid reference.